### ISSUE TITLE
Question on data

### ISSUE THREAD

**Issue by emelltech** [yashveersinghsohi/Statistical_Modeling_for_Time_Series_Forecasting#1]
Hi
I have a question on the contents of pred2_df.
When printing the tail the last date is 2019-08-30
Is the actual prediction for the next day (2019-08-31) contained in the spx_preds column ?

**Comment by yashveersinghsohi** [comment]
Hi.
I believe you are referring to a variable in this notebook [here](https://github.com/yashveersinghsohi/Statistical_Modeling_for_Time_Series_Forecasting/blob/master/Prices%20Models/ARMA%20Model%20for%20SPX%20Prices.ipynb)

I have explained the columns of this dataframe below. Let me know if any further clarification is needed -

- **spx:** This column stores the original dataset from the `data.csv` file's `spx` column exactly.
- **spx_1:** This column stores the values of `spx` column, but lagged by one timestep. That means that the the value stored in `spx_1` at `2019-08-30` is the value observed in `spx` at `2019-08-29`.
- **model_preds:** The ARMA model is not fit directly on `spx`. First, the data is converted to a stationary series, by applying log transform followed by successive differencing. Then the model is fit on this transformed `spx` series. `model_preds` store the predictions of the model on the transformed `spx`. So a value here on `2019-08-30` indicates the predicted value of transformed `spx` on `2019-08-30`.
- **model_preds_lower & model_preds_upper**: These 2 columns contain the upper and lower limits for `model_preds`.
- **model_preds_exp, model_preds_lower_exp, & model_preds_upper_exp:** Every value in these columns is calculated by taking the exponent of the corresponding value in the `model_preds`, `model_preds_lower`, and `model_preds_upper` columns respectively.
- **spx_preds:** This stores the actual predictions for the dataset. The predictions in `model_preds` are for the transformed `spx`. These are converted to predictions for original `spx` by using the following formula: ![y_t=y_{t-1}e^{y\_new_t}](http://www.sciweavers.org/tex2img.php?eq=y_t%3Dy_%7Bt-1%7De%5E%7By%5C_new_t%7D&bc=White&fc=Black&im=jpg&fs=12&ff=arev&edit=). So if the explanation for the above columns is clear, then the values in this formula would be:
  - ![y_t](http://www.sciweavers.org/tex2img.php?eq=y_t&bc=White&fc=Black&im=jpg&fs=12&ff=arev&edit=): `spx_preds` (This is to be calculated)
  - ![y_{t-1}](http://www.sciweavers.org/tex2img.php?eq=y_%7Bt-1%7D&bc=White&fc=Black&im=jpg&fs=12&ff=arev&edit=): `spx_1`. Values observed one time stamp previously.
  - ![y\_new_t](http://www.sciweavers.org/tex2img.php?eq=y%5C_new_t&bc=White&fc=Black&im=jpg&fs=12&ff=arev&edit=): `model_preds`. Predictions on transformed `spx`
  - ![e^{y\_new_t}](http://www.sciweavers.org/tex2img.php?eq=e%5E%7By%5C_new_t%7D&bc=White&fc=Black&im=jpg&fs=12&ff=arev&edit=): `model_preds_exp`
- **spx_preds_upper & spx_preds_lower:** Upper and lower limits on `spx_preds`, calculated using the same formula as above, but using the `model_preds_upper` and `model_preds-lower` respectively in place of `model_preds` in ![y\_new_t](http://www.sciweavers.org/tex2img.php?eq=y%5C_new_t&bc=White&fc=Black&im=jpg&fs=12&ff=arev&edit=).

To sum up, the `spx_preds` column on `2019-08-30` will contain the predictions for `spx` value on `2019-08-30`. To calculate this, the `spx` value from one time stamp ago (`spx_1`), and the exponent of the model's predictions of transformed `spx` for `2019-08-30` (`model_preds_exp`) are used.

Hope this helps.
